app/services/message_channel_service.rb

```diff
@@ -20,27 +20,25 @@ class UnknownChannelError < StandardError
                          discussion_description_edited
                          motion_description_edited)
 
-  def self.subscribe_to(user:, channel: )
-    raise AccessDeniedError.new unless can_subscribe?(user: user, channel: channel)
-    PrivatePub.subscription(channel: channel, server: ENV['FAYE_URL'])
+  def self.subscribe_to(user:, model:)
+    raise AccessDeniedError.new unless can_subscribe?(user: user, model: model)
+    PrivatePub.subscription(channel: channel_for(model), server: ENV['FAYE_URL'])
   end
 
-  def self.can_subscribe?(user:, channel:)
-    key = channel_key(channel)
-    case channel_type(channel)
-    when 'group'         then user.ability.can? :see_private_content, Group.find(key)
-    when 'discussion'    then user.ability.can? :show, Discussion.find(key)
-    when 'notifications' then key.to_i == user.id
-    else                      raise UnknownChannelError.new
-    end
+  def self.subscribe_to_notifications_for(user)
+    PrivatePub.subscription(channel: "/notifications/#{user.id}")
   end
 
-  def self.channel_type(channel)
-    /\/(\w+)-(\w+)/.match(channel)[1]
+  def self.can_subscribe?(user:, model:)
```

Comment:
At some point I want to be able to pass in a User here, with
```
case User then user.ability.can? :get_notifications_for, model
```

but I'll have to change the channel name to '/user-#{user_id}' to do that and didn't want to tread that just yet.